src/main/java/io/github/nambach/excelutil/core/ReaderConfig.java

```diff
@@ -281,6 +282,16 @@ public List<T> readSheet(InputStream stream, String sheetName) {
         }
     }
 
+    /**
+     * Read data from Excel and convert to stream of data directly.
+     *
+     * @param stream byte stream
+     * @return stream of DTO
+     */
+    public Stream<T> readSheetToStream(InputStream stream, String sheetName) {
```

Comment:
Could you describe the reason why we need this method when it could be simply derived from the original readSheet?

Reply:
Pardon me if I confused you, this line https://github.com/nambach/ExcelUtil/blob/18353a46333dfa6fcb9ac322a8b621d3921ca319/src/main/java/io/github/nambach/excelutil/core/ReaderConfig.java#L274
should be `public Result<T> readSheet(InputStream stream, String sheetName)` instead. That's my mistake 😅. The `read*` method should only return `Result<T>` since it will contain the validation errors:

```java
Result<Book> books = readerConfig.readSheet(stream);

// Check errors
if (books.hasErrors()) {
    for (RowError row : books.getErrors()) {
        // Handle errors
    }
}
```

You can read more on this https://github.com/nambach/ExcelUtil/wiki/Reading-Validation.